### ISSUE TITLE
Showing Train and Eval loss

### ISSUE THREAD

**Issue by mtortoli** [UKPLab/sentence-transformers#719]
Hi,
is it possible to view the loss value during training? And also the eval loss. I looked in the code if there is some method that returns the numeric value of the loss, but i couldn't find anything about it.

Thanks!

**Comment by nreimers** [comment]
Hi
You can add evaluators to the fit method: These are call every x training steps and can compute you relevant metrices, like accuracy on a dev set.

See the training examples how to use these evaluator.

Computing / reporting training loss is currently not implemented.

**Comment by mtortoli** [comment]
Thank you very much!

**Comment by hchings** [comment]
@nreimers 
Our team has the need to plot/report the loss curve either during or after the training.
I know you said it's currently not implemented, but is there any suggested approach to get the loss value of each epoch or any workaround you can point us to?
We're looking for a cleaner solution that does not require us to change `def fit()` in `SetenceTransformer.py`.
Thanks for this great package. It's been awesome and this is the only missing piece we wish it has.

@mtortoli Could you share your solution if any? Thanks

**Comment by nreimers** [comment]
Hi @hchings
You could create a custom evaluator, which computes the loss on some data, and pass it to the fit method. 

Another option would be to create a custom loss, that outputs the loss for you on the train set, and to train with this loss method.
